pkg/service/aws/ecr.go

```diff
@@ -0,0 +1,54 @@
+package aws
+
+import (
+	"context"
+
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/pkg/errors"
+	proto "gitlab.com/netbook-devs/spawner-service/proto/netbookai/spawner"
+)
+
+func (a *AWSController) GetElasticRegistryAuth(ctx context.Context, req *proto.GetElasticRegistryAuthRequest) (*proto.GetElasticRegistryAuthResponse, error) {
+
+	session, err := NewSession(ctx, req.Region, req.GetAccountName())
+
+	if err != nil {
+		a.logger.Error(ctx, "can't start AWS session", "error", err)
+		return nil, errors.Wrap(err, "GetWorkspacesCost: failed to get aws session")
+	}
+
+	//get account id
+	account_id, err := session.getAccountId()
+
+	if err != nil {
+		a.logger.Error(ctx, "failed to get account id", "error", err)
+		return nil, errors.Wrap(err, "GetWorkspacesCost: failed to get accoutn id")
+	}
+
+	a.logger.Debug(ctx, "fetched accountId", "id", account_id)
+
+	//get ecr token
+	ecrClient := session.getEcrClient()
+
+	// Doc : https://docs.aws.amazon.com/sdk-for-go/api/service/ecr/#ECR.GetAuthorizationToken
+	res, err := ecrClient.GetAuthorizationTokenWithContext(ctx, &ecr.GetAuthorizationTokenInput{
+		RegistryIds: []*string{&account_id},
```

Comment:
Were you able to test this? Seems weird that we are sending AWS account id as Registry Id

Reply:
yes, account Id is registryId in ECR context